src/main/kotlin/tech/glima/di/module/ServiceModule.kt

```diff
@@ -0,0 +1,15 @@
+package tech.glima.di.module
+
+import org.koin.dsl.module
+import tech.glima.service.CustomerService
+import tech.glima.service.CustomerServiceImpl
+
+object ServiceModule {
+
+    val module = module {
+
+        single<CustomerService> {
+            CustomerServiceImpl()
```

Comment:
é aqui que a magia acontece, 
```kotlin: 
single<CustomerService> {
   CustomerServiceImpl()
}
```
está definindo que uma instância de `CustomerServiceImpl` será provida sempre que uma `CustomerService` for solicitada.
neste caso está sendo fornecida a mesma instância (singleton) através do operador `single<T>`. Caso queira devolver uma instancia nova a cada requiisição use o operador `factory<T>`